pykafka/balancedconsumer.py

```diff
@@ -547,6 +561,10 @@ def _rebalance(self):
                         raise
                     log.info('Unable to acquire partition %s. Retrying', ex.partition)
                     time.sleep(i * (self._rebalance_backoff_ms / 1000))
+            if self._partitions_assigned_callback is not None:
+                self._partitions_assigned_callback(self, new_partitions)
+            if self._partitions_revoked_callback is not None:
+                self._partitions_revoked_callback(self, current_zk_parts - new_partitions)
```

Comment:
If the user wants to extract the `held_offsets` for the revoked partitions (to store them in their own custom offset store), I think this callback comes too late, because the internal consumer has at this point been replaced by the new one already.